seqr/views/apis/individual_api.py

```diff
@@ -787,20 +787,35 @@ def save_individuals_metadata_table_handler(request, project_guid, upload_file_i
 
 @login_and_policies_required
 def get_individual_rna_seq_data(request, individual_guid):
-    individual = Individual.objects.get(guid=individual_guid)
-    check_project_permissions(individual.family.project, request.user)
-    outlier_data = RnaSeqOutlier.objects.filter(sample__individual=individual, sample__is_active=True)
+    outlier_data = {'outliers': _get_rna_seq_data(RnaSeqOutlier, request, individual_guid)}
+    outlier_data.update({'spliceOutliers': _get_rna_seq_data(RnaSeqSpliceOutlier, request, individual_guid)})
 
-    rna_seq_data = {
-        data['geneId']: data for data in get_json_for_rna_seq_outliers(outlier_data)
-    }
-    genes_to_show = get_genes([gene_id for gene_id, data in rna_seq_data.items() if data['isSignificant']])
+    genes_to_show = get_genes({
+        gene_id for rna_data in outlier_data.values() for gene_id, data in rna_data.items() if data['isSignificant']
+    })
 
     return create_json_response({
-        'rnaSeqData': {individual_guid: {'outliers': rna_seq_data}},
+        'rnaSeqData': {individual_guid: {outlier_type: data for outlier_type, data in outlier_data.items() if data}},
         'genesById': genes_to_show,
+        'igvSamplesByGuid': {
+            igvs['sampleGuid']: igvs for igvs in get_json_for_samples(IgvSample.objects.filter(individual__guid=individual_guid))
+        },
     })
 
+
+def _get_rna_seq_data(model_cls, request, individual_guid):
+    individual = Individual.objects.get(guid=individual_guid)
+    check_project_permissions(individual.family.project, request.user)
```

Comment:
this code is shared between the 2 RNA models so it should be kept in the main function and only called once